qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,231 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+T1 Experiment class.
+"""
+
+from typing import List, Optional, Union, Tuple
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments import AnalysisResult
+from .analysis_functions import exp_fit_fun, curve_fit_wrapper
+
+
+class T1Analysis(BaseAnalysis):
+    """T1 Experiment result analysis class."""
+
+    # pylint: disable=arguments-differ, unused-argument
+    def _run_analysis(
+        self,
+        experiment_data,
+        t1_guess=None,
+        amplitude_guess=None,
+        offset_guess=None,
+        t1_bounds=None,
+        amplitude_bounds=None,
+        offset_bounds=None,
+        **kwargs,
+    ) -> Tuple[AnalysisResult, None]:
+        """
+        Calculate T1
+
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze
+            t1_guess (float): Optional, an initial guess of T1
+            amplitude_guess (float): Optional, an initial guess of the coefficient of the exponent
+            offset_guess (float): Optional, an initial guess of the offset
+            t1_bounds (list of two floats): Optional, lower bound and uper bound to T1
+            amplitude_bounds (list of two floats): Optional, lower bound and uper bound to the amplitude
+            offset_bounds (list of two floats): Optional, lower bound and uper bound to the offset
+            kwargs: Trailing unused function parameters
+
+        Returns:
+            The analysis result with the estimated T1
```

Comment:
What are the units on the estimated T1?  (what are the units on all of the above I guess as well?).

Reply:
You set the unit in the constructor of `T1Experiment`

Reply:
That feels like it could get messy since any code aggregating multiple results will have to look at the units of each result and fix them up to match?  (Or is what goes into the db normalized differently?)

Reply:
Why not define everything in SI?